pointcloud/basic_octree_utils.go

```diff
@@ -8,31 +8,40 @@ import (
 )
 
 // Creates a new LeafNodeEmpty.
-func newLeafNodeEmpty() basicOctreeNode {
+func newLeafNodeEmpty(parent *BasicOctree, depth int) basicOctreeNode {
 	octNode := basicOctreeNode{
 		children: nil,
+		parent:   parent,
 		nodeType: leafNodeEmpty,
 		point:    PointAndData{},
+		maxProb:  -1.,
+		depth:    depth,
 	}
 	return octNode
 }
 
 // Creates a new InternalNode with specified children nodes.
-func newInternalNode(tree []*BasicOctree) basicOctreeNode {
+func newInternalNode(tree []*BasicOctree, parent *BasicOctree, depth int) basicOctreeNode {
 	octNode := basicOctreeNode{
 		children: tree,
+		parent:   parent,
 		nodeType: internalNode,
 		point:    PointAndData{},
+		maxProb:  -1.,
+		depth:    depth,
 	}
 	return octNode
 }
 
 // Creates a new LeafNodeFilled and stores specified position and data.
-func newLeafNodeFilled(p r3.Vector, d Data) basicOctreeNode {
+func newLeafNodeFilled(p r3.Vector, d Data, parent *BasicOctree, depth int) basicOctreeNode {
 	octNode := basicOctreeNode{
 		children: nil,
+		parent:   parent,
 		nodeType: leafNodeFilled,
 		point:    PointAndData{P: p, D: d},
+		maxProb:  float64(d.Value()) / 100,
```

Comment:
[major] must check data for HasValue(), if its doesn't then assign to 1.0

Reply:
Do we want to error out if the value returned is >100 or <0?

Reply:
> Do we want to error out if the value returned is >100 or <0?

yes, I will add an error statement should value be >100 or <0.